optuna/pruners/successive_halving.py

```diff
@@ -46,6 +49,9 @@ class SuccessiveHalvingPruner(BasePruner):
             A parameter for specifying the minimum resource allocated to a trial
             (in the `paper <http://arxiv.org/abs/1810.05934>`_ this parameter is
             referred to as :math:`r`).
+            This parameter defaults to ``'auto'`` that means this pruner estimates an appropriate
+            minimum resource by the required number of a ``COMPLETE``
+            :class:`~optuna.structs.FrozenTrial`'s steps.
```

Comment:
I believe this descriptions is quite "internal" and could be a tiny bit more user friendly/descriptive. Something along.

> This parameter defaults to ``'auto'`` where the value is determined based on a heuristic that looks at the number of required steps for the first trial before completion.

Reply:
Positive.

I'll use your description.